Compilation error: exit status 1

Board: Adafruit QT PY RP2040
Arduino IDE Version: 2.3.7
RD2040 compile error seems to be hardware related compiles and runs on the Adafruit QT PY ESP32-S3
 ` Arduino\libraries\Adafruit_Si5351_Library\Adafruit_SI5351.cpp:56:1: error: expected unqualified-id before '/' token
   56 | /
      | ^
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:118:61: error: expected unqualified-id before '{' token
  118 | static inline unsigned char pgm_read_byte(const void *addr) {
      |                                                             ^
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h: In function 'short unsigned int pgm_read_word(const void*)':
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:123:29: error: expected primary-expression before '|' token
  123 |     return pgm_read_byte(a) | ( pgm_read_byte(a + 1) << 8 );
      |                             ^
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:123:54: error: expected primary-expression before '<<' token
  123 |     return pgm_read_byte(a) | ( pgm_read_byte(a + 1) << 8 );
      |                                                      ^~
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h: In function 'long unsigned int pgm_read_dword(const void*)':
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:127:29: error: expected primary-expression before '|' token
  127 |     return pgm_read_byte(a) | ( pgm_read_byte(a + 1) << 8 ) | ( pgm_read_byte(a + 2) << 16 ) | ( pgm_read_byte(a + 3) << 24 );
      |                             ^
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:127:54: error: expected primary-expression before '<<' token
  127 |     return pgm_read_byte(a) | ( pgm_read_byte(a + 1) << 8 ) | ( pgm_read_byte(a + 2) << 16 ) | ( pgm_read_byte(a + 3) << 24 );
      |                                                      ^~
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:127:86: error: expected primary-expression before '<<' token
  127 |     return pgm_read_byte(a) | ( pgm_read_byte(a + 1) << 8 ) | ( pgm_read_byte(a + 2) << 16 ) | ( pgm_read_byte(a + 3) << 24 );
      |                                                                                      ^~
Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/deprecated-avr-comp/avr/pgmspace.h:127:119: error: expected primary-expression before '<<' token
  127 |     return pgm_read_byte(a) | ( pgm_read_byte(a + 1) << 8 ) | ( pgm_read_byte(a + 2) << 16 ) | ( pgm_read_byte(a + 3) << 24 );
      |                                                                                                                       ^~
In file included from Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/cmath:45,
                 from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/math.h:36,
                 from  Arduino15/packages/rp2040/hardware/rp2040/5.4.4/ArduinoCore-API/api/ArduinoAPI.h:47:
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/ext/type_traits.h: At global scope:
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/ext/type_traits.h:164:35: error: 'constexpr const bool __gnu_cxx::__is_null_pointer' redeclared as different kind of entity
  164 |   __is_null_pointer(std::nullptr_t)
      |                                   ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/ext/type_traits.h:159:5: note: previous declaration 'template<class _Type> constexpr bool __gnu_cxx::__is_null_pointer(_Type)'
  159 |     __is_null_pointer(_Type)
      |     ^~~~~~~~~~~~~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/ext/type_traits.h:164:26: error: 'nullptr_t' is not a member of 'std'; did you mean 'nullptr_t'?
  164 |   __is_null_pointer(std::nullptr_t)
      |                          ^~~~~~~~~
In file included from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/sys/reent.h:14,
                 from Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/stdlib.h:18,
                 from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/cstdlib:79,
                 from Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/stdlib.h:36,
                 from  Arduino\libraries\Adafruit_Si5351_Library\Adafruit_SI5351.cpp:59:
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/lib/gcc/arm-none-eabi/14.3.0/include/stddef.h:458:29: note: 'nullptr_t' declared here
  458 |   typedef decltype(nullptr) nullptr_t;
      |                             ^~~~~~~~~
In file included from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/bits/stl_pair.h:60,
                 from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/bits/stl_algobase.h:64,
                 from Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/bits/specfun.h:43,
                 from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/cmath:3906:
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:666:33: error: 'nullptr_t' is not a member of 'std'; did you mean 'nullptr_t'?
  666 |     struct is_null_pointer<std::nullptr_t>
      |                                 ^~~~~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/lib/gcc/arm-none-eabi/14.3.0/include/stddef.h:458:29: note: 'nullptr_t' declared here
  458 |   typedef decltype(nullptr) nullptr_t;
      |                             ^~~~~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:666:42: error: template argument 1 is invalid
  666 |     struct is_null_pointer<std::nullptr_t>
      |                                          ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:670:48: error: template argument 1 is invalid
  670 |     struct is_null_pointer<const std::nullptr_t>
      |                                                ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:674:51: error: template argument 1 is invalid
  674 |     struct is_null_pointer<volatile std::nullptr_t>
      |                                                   ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:678:57: error: template argument 1 is invalid
  678 |     struct is_null_pointer<const volatile std::nullptr_t>
      |                                                         ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1429:37: error: 'size_t' is not a member of 'std'; did you mean 'size_t'?
 1429 |     : public integral_constant<std::size_t, alignof(_Tp)>
      |                                     ^~~~~~
In file included from  Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/stdlib.h:16:
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/lib/gcc/arm-none-eabi/14.3.0/include/stddef.h:229:23: note: 'size_t' declared here
  229 | typedef __SIZE_TYPE__ size_t;
      |                       ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1429:57: error: template argument 1 is invalid
 1429 |     : public integral_constant<std::size_t, alignof(_Tp)>
      |                                                         ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1429:57: note: invalid template non-type parameter
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1438:37: error: 'size_t' is not a member of 'std'; did you mean 'size_t'?
 1438 |     : public integral_constant<std::size_t, 0> { };
      |                                     ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/lib/gcc/arm-none-eabi/14.3.0/include/stddef.h:229:23: note: 'size_t' declared here
  229 | typedef __SIZE_TYPE__ size_t;
      |                       ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1438:46: error: template argument 1 is invalid
 1438 |     : public integral_constant<std::size_t, 0> { };
      |                                              ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1438:46: note: invalid template non-type parameter
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1440:26: error: 'std::size_t' has not been declared
 1440 |   template<typename _Tp, std::size_t _Size>
      |                          ^~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1441:21: error: '_Size' was not declared in this scope
 1441 |     struct rank<_Tp[_Size]>
      |                     ^~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1441:27: error: template argument 1 is invalid
 1441 |     struct rank<_Tp[_Size]>
      |                           ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1442:37: error: 'size_t' is not a member of 'std'; did you mean 'size_t'?
 1442 |     : public integral_constant<std::size_t, 1 + rank<_Tp>::value> { };
      |                                     ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/lib/gcc/arm-none-eabi/14.3.0/include/stddef.h:229:23: note: 'size_t' declared here
  229 | typedef __SIZE_TYPE__ size_t;
      |                       ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1442:65: error: template argument 1 is invalid
 1442 |     : public integral_constant<std::size_t, 1 + rank<_Tp>::value> { };
      |                                                                 ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1442:65: note: invalid template non-type parameter
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1446:37: error: 'size_t' is not a member of 'std'; did you mean 'size_t'?
 1446 |     : public integral_constant<std::size_t, 1 + rank<_Tp>::value> { };
      |                                     ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/lib/gcc/arm-none-eabi/14.3.0/include/stddef.h:229:23: note: 'size_t' declared here
  229 | typedef __SIZE_TYPE__ size_t;
      |                       ^~~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1446:65: error: template argument 1 is invalid
 1446 |     : public integral_constant<std::size_t, 1 + rank<_Tp>::value> { };
      |                                                                 ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:1446:65: note: invalid template non-type parameter
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2086:26: error: 'std::size_t' has not been declared
 2086 |   template<typename _Tp, std::size_t _Size>
      |                          ^~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2087:30: error: '_Size' was not declared in this scope
 2087 |     struct remove_extent<_Tp[_Size]>
      |                              ^~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2087:36: error: template argument 1 is invalid
 2087 |     struct remove_extent<_Tp[_Size]>
      |                                    ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2099:26: error: 'std::size_t' has not been declared
 2099 |   template<typename _Tp, std::size_t _Size>
      |                          ^~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2100:35: error: '_Size' was not declared in this scope
 2100 |     struct remove_all_extents<_Tp[_Size]>
      |                                   ^~~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2100:41: error: template argument 1 is invalid
 2100 |     struct remove_all_extents<_Tp[_Size]>
      |                                         ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2171:12: error: 'std::size_t' has not been declared
 2171 |   template<std::size_t _Len>
      |            ^~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2176:30: error: '_Len' was not declared in this scope
 2176 |         unsigned char __data[_Len];
      |                              ^~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2194:12: error: 'std::size_t' has not been declared
 2194 |   template<std::size_t _Len, std::size_t _Align =
      |            ^~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2194:30: error: 'std::size_t' has not been declared
 2194 |   template<std::size_t _Len, std::size_t _Align =
      |                              ^~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2195:55: error: '_Len' was not declared in this scope
 2195 |            __alignof__(typename __aligned_storage_msa<_Len>::__type)>
      |                                                       ^~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2195:59: error: template argument 1 is invalid
 2195 |            __alignof__(typename __aligned_storage_msa<_Len>::__type)>
      |                                                           ^
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2202:30: error: '_Len' was not declared in this scope
 2202 |         unsigned char __data[_Len];
      |                              ^~~~
 Arduino15/packages/rp2040/tools/pqt-gcc/4.1.0-1aec55e/arm-none-eabi/include/c++/14.3.0/type_traits:2203:44: error: '_Align' was not declared in this scope
 2203 |         struct __attribute__((__aligned__((_Align)))) { } __align;
      |                                            ^~~~~~
exit status 1

Compilation error: exit status 1`

--- a/Adafruit_SI5351.cpp
+++ b/Adafruit_SI5351.cpp
@@ -49,7 +49,8 @@
 #if defined(__AVR__)
 #include <avr/pgmspace.h>
 #include <util/delay.h>
-#elif defined(ESP8266) || defined(ESP32)
+#elif defined(ESP8266) || defined(ESP32) || \
+    (defined(ARDUINO_ARCH_RP2040) && !defined(__MBED__))
 #include "pgmspace.h"
 #else
 #define pgm_read_byte(addr) \

--- a/examples/si5351/si5351.ino
+++ b/examples/si5351/si5351.ino
@@ -10,6 +10,8 @@ Adafruit_SI5351 clockgen = Adafruit_SI5351();
 void setup(void)
 {
   Serial.begin(9600);
+  while (!Serial)
+    delay(10); // wait for native USB (e.g. RP2040, SAMD)
   Serial.println("Si5351 Clockgen Test"); Serial.println("");
 
   /* Initialise the sensor */

